common_knowledge/Federation.md

```diff
@@ -0,0 +1,38 @@
+# Federation (WIP)
+
+The Commonwealth forum implements federation capabilities using
+libraries from [Canvas](https://docs.canvas.xyz).
+
+This means that forum actions are signed using the users' wallet
+address, with one level of delegation (the user authorizes a session
+key when they log in).
+
+## Testing federation locally
+
+From packages/commonwealth, run `pnpm start` to start the
+application. In the log output, look for:
+
+```
+canvas: started libp2p with multiaddrs [
+    /ip4/127.0.0.1/tcp/8090/ws/p2p/12D3KooWPbkvKDxziibMP3cxrQP7NNnZCDNfj2cT3Ej3TH23wWBS
+]
+```
+
+In another terminal, use the libp2p multiaddr to start a node:
+
+```
+canvas run libs/shared/src/canvas/runtime/contract.ts --bootstrap-list /ip4/127.0.0.1/tcp/8090/ws/p2p/12D3KooWPbkvKDxziibMP3cxrQP7NNnZCDNfj2cT3Ej3TH23wWBS
```

Comment:
I cant actually get this step to work, so I have downloaded the canvas cli like:

```bash
 npm install -g @canvas-js/cli
```

And then tried running this line but get:
`Unknown arguments: bootstrap-list, bootstrapList`

Trying to change it to --bootstrap, I get:
```
Contract files must match *.canvas.js
```

For reference I tried this with the latest canvas cli as well as the cli version 0.10.8. Maybe you can add to the doc the specific version needed to install?

Reply:
I had cut another release to get some of the features in the Canvas CLI, but the CLI should work now. Peering with the CLI *might* not work depending on your local setup... I think it's probably best to focus on the Commonwealth side rather than trying to QA peering, since strictly speaking that's in the next PR (I started adding the docs to this PR as a work in progress but they are not really complete.) 

I'm working on merging up to `master` which now includes the new API, which is holding up getting actions to propagate from the Canvas side to the Common side.